From 4a92f79321ce925eb09dc32d526606035bd55bdb Mon Sep 17 00:00:00 2001
From: Kevin Tang <zhikait@codeaurora.org>
Date: Wed, 13 Jan 2016 13:14:03 -0800
Subject: [PATCH 55/56] combo issue when location reported by providers do not
 come with extras

LocationManagerService tries to create an extra bundle and stick it
into the location object, but it is not doing it correctly. As a
result, combo drops report.

Change-Id: I6f5886dd7f14819ecc7a4ed96c867a0ec486faf7
CRs-Fixed: 962746
---
 .../java/com/android/server/LocationManagerService.java  | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/LocationManagerService.java b/services/core/java/com/android/server/LocationManagerService.java
index 0adcab1e2d2c..0b55ee71b0c3 100644
--- a/services/core/java/com/android/server/LocationManagerService.java
+++ b/services/core/java/com/android/server/LocationManagerService.java
@@ -3113,11 +3113,8 @@ public class LocationManagerService extends ILocationManager.Stub {
 
         Bundle extras = location.getExtras();
         boolean isBeingScreened = false;
-        if (extras == null) {
-            extras = new Bundle();
-        }
 
-        if (!extras.containsKey(mComboNlpReadyMarker)) {
+        if (extras == null || !extras.containsKey(mComboNlpReadyMarker)) {
             // see if Combo Nlp is a passive listener
             ArrayList<UpdateRecord> records =
                 mRecordsByProvider.get(LocationManager.PASSIVE_PROVIDER);
@@ -3126,6 +3123,10 @@ public class LocationManagerService extends ILocationManager.Stub {
                     if (r.mReceiver.mIdentity.mPackageName.equals(mComboNlpPackageName)) {
                         if (!isBeingScreened) {
                             isBeingScreened = true;
+                            if (extras == null) {
+                                location.setExtras(new Bundle());
+                                extras = location.getExtras();
+                            }
                             extras.putBoolean(mComboNlpScreenMarker, true);
                         }
                         // send location to Combo Nlp for screening
-- 
2.17.1

